test/simulator/timestamp/src/session_simulator.cpp

```diff
@@ -59,3 +60,48 @@ session_simulator::commit_transaction()
 
     _txn_running = false;
 }
+
+void
+session_simulator::set_commit_timestamp(const uint64_t &ts)
+{
+    _commit_ts = ts;
+}
+
+void
+session_simulator::set_durable_timestamp(const uint64_t &ts)
+{
+    _durable_ts = ts;
+}
+
+void
+session_simulator::set_prepare_timestamp(const uint64_t &ts)
+{
+    _prepare_ts = ts;
+}
+
+void
+session_simulator::set_read_timestamp(const uint64_t &ts)
+{
+    _read_ts = ts;
+}
+
+int
+session_simulator::timestamp_transaction_uint(const std::string &ts_type, const uint64_t &ts)
+{
+    /* Zero timestamp is not permitted. */
+    if (ts == 0) {
```

Comment:
Could this be an assertion instead? It's fine as it is but sometimes we do asserts, sometimes `if` and a message.

Reply:
That's true, I think an assert would be more consistent.

Reply:
Actually, sorry, we want to match up the return value from the simulator with the call log entry generated from wiredtiger so in the case of having a 0 timestamp we can check that `EINVAL` is returned.